### PR TITLE
[feat] 简单 API, 题目配置重载, 文件夹上传

### PR DESCRIPTION
**Pull request by YooLc**

### CODE REVIEW

File: judge.go [hunk 1]
```diff
@@ -19,28 +23,28 @@ import (
 )
 
 type JudgeResult struct {
-	Success bool
+	Success bool `json:"success"`
```
Comment (mrhaoxx) [inline judge.go]: 这里似乎是 breaking-change 啊

File: judge.go [hunk 2]
```diff
@@ -71,32 +75,33 @@ type SubmitHash struct {
...
 type SubmitCtx struct {
-	ID      string `gorm:"primaryKey"`
-	User    string
-	Problem string
+	ID      string `gorm:"primaryKey" json:"id"`
+	User    string `json:"user"`
+	Problem string `json:"problem"`
 
-	problem *Problem
+	problem *Problem `json:"-"`
```
Comment (mrhaoxx) [inline judge.go]: 不需要加，本身就没有导出

Comment (mrhaoxx) [inline judge.go]: 不需要加，本身就没有导出

File: judge.go [hunk 3]
```diff
@@ -71,32 +75,33 @@ type SubmitHash struct {
...
+	WorkflowResults WorkflowResults `json:"workflow_results"`
+	JudgeResult     JudgeResult     `json:"judge_result"`
 
-	RealWorkdir string
+	RealWorkdir string `json:"-"`
 
-	running  chan struct{}
-	Userface Userface
+	running  chan struct{} `json:"-"`
+	Userface Userface      `json:"-"`
```
Comment (mrhaoxx) [inline judge.go]: userface保存了日志，看看前端需不需要展示，可以参考db.go里的序列化方法

File: judge.go [hunk 4]
```diff
@@ -216,34 +221,37 @@ workdir_created:
...
+				}
+				return nil
 			})
-
-			ctx.Userface.Println("	*", aurora.Yellow(submit.Path), ":", aurora.Blue(hash))
+			if err != nil {
+				ctx.SetStatus("failed").SetMsg("failed to copy submit directory " + strconv.Quote(submit.Path)).Update()
+				ctx.Userface.Println("	*", aurora.Yellow(submit.Path), ":", aurora.Red("failed"))
+				return
+			}
```
Comment (mrhaoxx) [inline judge.go:253]: 提交dir现在有用到吗？没用的话就不要加，感觉不太好

File: database/init.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	})
+	return db
+}
+
+func GetDB() *gorm.DB {
+	once.Do(func() {
+		log.Fatal().Msg("Used database before initialization")
+	})
+	return db
+}
```
Comment (mrhaoxx) [inline database/init.go:31]: 分离的意义何在（

File: http.go [hunk 6]
```diff
@@ -1,6 +1,81 @@
...
+	c.JSON(200, gin.H{
+		"code":    0,
+		"message": "success",
+		"data": gin.H{
+			"total":   total,
+			"submits": submits,
+		},
+	})
+}
+
```
Comment (mrhaoxx) [inline http.go:47]: 鉴权？

File: judge.go (hunk 1 above)
Comment (YooLc) [reply judge.go]: > To unmarshal JSON into a struct, Unmarshal matches incoming object keys to the keys used by [Marshal](https://pkg.go.dev/encoding/json#Marshal) (either the struct field name or its tag), 
**preferring an exact match but also accepting a case-insensitive match**. By default, object keys which don't have a corresponding struct field are ignored (see [Decoder.DisallowUnknownFields](https://pkg.go.dev/encoding/json#Decoder.DisallowUnknownFields) for an alternative).

不影响倒是（
我改回去吧

**Pull request merged**